### ISSUE TITLE
A way to turn off "Read Bookmark Quote" highlight bar

### ISSUE THREAD

**Issue by SomewhatDamaged** [knighty/fimfiction-issues#30]
![This thing](https://goo.gl/obyseF)

It would be great if there was a way to make it not show up while clicking within a story. The features it gives are nice, but if I just want to relax and read it distracts from the actual words quite a lot.

**Comment by FM-96** [comment]
I agree. In addition, it also kind of messes with text selection on mobile Chrome.

I think it would be great if a switch to turn it off was added to the local (aka device specific) settings.

**Comment by knighty** [comment]
I'm not convinced this is an issue that needs altering. Most of it boils down to "I compulsively click things" which I don't think is a good idea to alter a feature for. If there was an option it'd mean you have no way to start text to speech or bookmark things.

**Comment by SomewhatDamaged** [comment]
The main issue I have is when I click back to the window to focus it (from doing something else). The click to focus the window causes this to trigger. Could it maybe be an option to change it from single click to double click? This would make it still be usable, but less likely to trigger from every odd click on the window.

**Comment by FM-96** [comment]
@knighty 
Well, that's why we think it should be an _option_. I cannot speak for everybody, but I personally value my ability to properly select text on my phone higher than my ability to start text to speech (I will probably never use that) or my ability to place manual bookmarks (doesn't the bookmarking happen automatically, anyway?).

The quote feature could potentially be useful to me, but so far I have used it intentionally 0 times, but I have accidentally clicked it twice (once while trying to remove the yellow box and once just by laying my finger on the screen), which autoscrolls me to the very bottom of the page. That... gets old really fast, to be honest.

And if someone does want to use text to speech or place bookmarks, they can just leave it turned on. It seems like an everybody-wins situation to me.

**Comment by knighty** [comment]
@SomewhatDamaged 
@FM-96 
I've checked in what I consider solutions to both problems. If you have text selected, clicking will not select a paragraph until the text is unselected, and going to another window and clicking back also won't select one immediately either.

**Comment by SomewhatDamaged** [comment]
@knighty 
Beautiful, that solves the two issues I had with it, and still leaves it perfectly usable.

**Comment by FM-96** [comment]
@knighty 
I'm a bit confused. How does this solve the text selection issue on mobile Chrome? Obviously I _don't_ currently have text selected if I want to select text...

Edit: Not triggering the yellow box while text is selected was a good call, though.

**Comment by knighty** [comment]
@FM-96 How do you select text? Holding to select for me doesn't do any kind of paragraph selection.... I thought you meant the case where you had it selected and wanted to deselect it.

**Comment by FM-96** [comment]
@knighty 
Ah, sorry. I should have been more specific.

On mobile Chrome, there are two ways to select text. A long press selects just the word you are pressing on (this works just fine), and a short tap uses a sort of "smart select". So if a story has the sentence "The next day, Twilight Sparkle went home." and I do a short tap on "Sparkle", Chrome correctly guesses that I want "Twilight Sparkle" selected.

Since this smart select is both quicker and also guesses correctly almost every time, I (and I would assume most mobile Chrome users) have come to use it as the default way of selecting text. By now, my muscle memory makes me instinctively single tap on a word if I want to select it.

Now though, Fimfiction intercepts that short tap and instead of highlighting the words, I get the yellow box.

I also want to note that since this method still works on almost every other website, it's not really a "I'll get used to it" situation. Hard to unlearn something that only doesn't work on one website.

(If my memory serves me right, that was also how one of the aforementioned accidental pressings of the "Quote" button ended up happening: Try to select word, get yellow box instead, try to close it, butterfinger it and press "Quote", end up at the bottom of the page.)

**Comment by knighty** [comment]
Ah, I have smart select off because it used to drive me mad. While appreciate your particular usage causes annoyance here I'm afraid I'm going to have to say this probably isn't going to change. I'm going to leave the Issue open in case I think of something to aid this problem, but it won't be happening in the very near future. Sorry

**Comment by ssokolow** [comment]
> Most of it boils down to "I compulsively click things" which I don't think is a good idea to alter a feature for.

I see it more as "This feature is a massive departure from a fundamental aspect of the behaviour common to all other applications and websites the user interacts with". (Any Human-Computer Interaction professional will tell you that's a massive no-no.)

Why not make it a `:hover` hamburger button which appears in right margin of the paragraph instead? That's more in-line with what other applications do for this sort of interaction, so it would be better both on consistency and on people's actual complaints.

(And, given how phones handle `:hover`, it'd still be accessible by tapping on the paragraph, just with an extra tap to open the hamburger.)

**Comment by andrewmyles** [comment]
Perhaps you haven't realised that, but there are non-English people reading fics on this site. And, surprisingly, they don't always know all the words. So, what's the easiest way to check it? Well, double click a word, to copy and paste it to a translator. Except that triggers the silly yellow box. Seriously, add an off-switch button in options. 

Oh, and if you could also add an option to turn off automatic bookmarking, that would be lovely. I hate when I check a story to see it, and it turns out it's not my cup of tea, but fimfiction insists that I should back to reading it.
